src/linters/spacing.js

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable import/no-commonjs */
+
+const fs = require('fs');
+const path = require('path');
+const stylelint = require('stylelint');
+
+const spacing = fs.readFileSync(path.join(__dirname, '..', 'styles', 'spacing.css'), 'UTF-8');
+const cssVarRegex = /--[a-z0-9-]+/g;
+const validSpacingSettings = spacing.match(cssVarRegex);
+
+const cssValueRegex = /[0-9]+px/g;
+const valueKeys = spacing.match(cssValueRegex);
+
+const ruleName = 'mds/spacing';
+const url = 'https://mavenlink.github.io/design-system/master/#/Brand%20Identity/Spacing';
+
+function zip(obj, value, index) {
+  return { ...obj, [value]: validSpacingSettings[index] };
+}
+const fixmap = valueKeys.reduce(zip, {});
+
+const messages = stylelint.utils.ruleMessages(ruleName, {
+  rejected: `To promote design consistency: use MDS spacing variables for margin and padding.
+This rule can always be auto fixed with the --fix option.
+Ie. ${JSON.stringify(fixmap)} For more info, see: ${url}.
+If you are using a spacing value that is close to the MDS variable value please consider using the MDS value.`,
+});
+
+const properties = [
+  'margin',
+  'margin-top',
+  'margin-right',
+  'margin-bottom',
+  'margin-left',
+  'padding',
+  'padding-top',
+  'padding-right',
+  'padding-bottom',
+  'padding-left',
+];
+
+const fixes = [];
+
+module.exports = stylelint.createPlugin(ruleName, (primary, secondary, context) => {
+  return (root, result) => {
+    root.walkDecls((declaration) => {
+      const property = declaration.prop;
+
+      if (properties.includes(property)) {
+        const valueTokens = declaration.value.split(' ');
+
+        const includesCalc = declaration.value.includes('calc');
+        if (includesCalc) {
+          return;
+        }
+
+        const validSpacingValue = valueTokens.every((token) => {
+          const includesSpacingVariable = validSpacingSettings.find((spacingVariable) => {
+            return token.includes(`var(${spacingVariable})`);
+          });
+          return includesSpacingVariable || token === '0' || !Object.keys(fixmap).includes(token);
+        });
+
+        if (!validSpacingValue) {
```

Comment:
We normally prefer to use positive assertions in the `if` when there is an `else`. 

For this instance, is this variable misnamed? `!!validSpacingValue` is coded to report a violation. ?

Reply:
We could also have the function return the opposite of what's given, perhaps using `some` instead of `every`, and then rename the variable to `invalidSpacingValue`